dpytools/http_clients/base.py

```diff
@@ -0,0 +1,51 @@
+import backoff
+import requests
+from requests.exceptions import HTTPError
+import logging
+
+
+# Function to log retry attempts
+def log_retry(details):
+    logging.error(f"Request failed, retrying... Attempt #{details['tries']}")
+
+
+class BaseHttpClient:
+    # Initialize HttpClient with a backoff_max value
+    def __init__(self, backoff_max=30):
+        self.backoff_max = backoff_max
+
+    # GET request method with exponential backoff
+    @backoff.on_exception(
+        backoff.expo, 
+        HTTPError,
+        max_time=30,  
+        on_backoff=log_retry
+    )
+    def get(self, url, *args, **kwargs):
```

Comment:
can you add a docstring and include a link to the request.get() documentation. That way a developer using the client can lookup what args and kwargs they can use without us having to explicitly document them.

Also, can you add a type hint this that it's return a requests `Response` object please. Future devs will know what they've for to work with then.